children/tests/test_api.py

```diff
@@ -178,6 +188,60 @@ def test_children_query_staff_user(snapshot, staff_api_client):
     snapshot.assert_match(executed)
 
 
+ADD_CHILD_MUTATION = """
+mutation addChild($input: AddChildMutationInput!) {
+  addChild(input: $input) {
+    child {
+      firstName
+      lastName
+      birthdate
+      postalCode
+    }
+  }
+}
+"""
+
+
+ADD_CHILD_VARIABLES = {
+    "input": {
+        "firstName": "Pekka",
+        "lastName": "Perälä",
+        "birthdate": "2020-11-11",
+        "postalCode": "00820",
+        "relationship": {"type": "PARENT"},
+    }
+}
+
+
+@pytest.mark.django_db
+def test_add_child_mutation(snapshot, guardian_api_client):
+    executed = guardian_api_client.execute(
+        ADD_CHILD_MUTATION, variables=ADD_CHILD_VARIABLES
+    )
+
+    snapshot.assert_match(executed)
+
+    child = Child.objects.last()
+    assert_child_matches_data(child, ADD_CHILD_VARIABLES["input"])
+
+    relationship = Relationship.objects.get(
+        guardian=guardian_api_client.user.guardian, child=child
+    )
+    assert_relationship_matches_data(
+        relationship, ADD_CHILD_VARIABLES["input"]["relationship"]
+    )
+
+
+@pytest.mark.django_db
+def test_add_child_mutation_birthdate_required(snapshot, user_api_client):
+    variables = deepcopy(UPDATE_CHILD_VARIABLES)
+    variables["input"].pop("postalCode")
```

Comment:
1. test name kinda suggests that you wanted to pop `birthdate` 😅 
2. not sure why poping `postalCode` fails the mutation though, since it's non-required 🤔

Reply:
Yeah, this test was actually flawed in a couple of ways, good thing you spotted it :)

This actually showed that just checking for "errors" to exists in the response isn't a very optimal way in this kind of tests because of false positives, so gotta figure out a better way for these in the future (but not in this PR :) )

Reply:
heh, alright!